### ISSUE TITLE
super-init-not-called inconsistent behavior

### ISSUE THREAD

**Issue by jakeleventhal** [PyCQA/pylint#3072]
I'm getting non-deterministic results when running pylint with the `super-init-not-called` rule.

I tried to create the most basic example I could find and came up with this:
```python
from collections import MutableMapping

class MySpecialDict(MutableMapping):
    def __init__(self, *args, **kwargs):
        pass
    def __getitem__(self, key):
        pass
    def __setitem__(self, key, value):
        pass
    def __delitem__(self, key):
        pass
    def __iter__(self):
        pass
    def __len__(self):
        pass
```

When I run pylint (1.9.5 - I need 2/3 backwards compatibility), I indeterminately get 10/10 and an error that says ` __init__ method from base class 'MutableMapping' is not called (super-init-not-called)`.

Sometimes it works, sometimes it doesn't. I am calling pylint via the CLI with `pylint file.py`

**Comment by PCManticore** [comment]
Thanks for the report! I can reproduce this issue in the sense that it never triggers for the code you pasted. Definitely a bug somewhere.

**Comment by PCManticore** [comment]
After some investigation, it turns out this is expected. This check is emitted only when there is a non-object `__init__` somewhere in the MRO. That is, all `__init__` that are custom made will trigger this error, but if the only `__init__` in MRO belongs to `object`, we do not trigger this error. This seems to be the case for `MutableMapping` as well. 

Not sure why it triggers on Pylint 1.9.X but I wonder if it's a difference between Python 2 and 3, with the latter having `object` by default as a base class.
